storage/src/node/branch.rs

```diff
@@ -91,6 +94,23 @@ pub(crate) trait ReadSerializable: Read {
 
 impl<T: Read> ReadSerializable for T {}
 
+#[cfg(feature = "ethhash")]
+pub(crate) struct NodeRefWithHashMut<'a> {
+    pub node_ref: Box<dyn Deref<Target = Node> + 'a>,
+    pub hash: &'a mut HashType,
+}
+
+#[cfg(feature = "ethhash")]
+impl std::fmt::Debug for NodeRefWithHashMut<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "NodeRefWithHashMut {{ node_ref: {:?}, hash: {:?} }}",
+            **self.node_ref, self.hash
+        )
+    }
```

Comment:
You can avoid this by creating a new trait:
```
#[cfg(feature = "ethhash")]
trait NodeRef: std::ops::Deref<Target = Node> + std::fmt::Debug {}
```

Then, change `node_ref` to `Box<dyn NodeRef + 'a>`

Reply:
It doesn't seem to work cuz `&Node` is not `std::fmt::Debug` :(

Reply:
I don't understand what you mean by this. I made the change and pushed it to https://github.com/ava-labs/firewood/tree/rkuris/qusuyan/hash-helper-reorg and it compiles fine. I didn't run all the tests though.

You can see the exact diff I applied here: https://github.com/ava-labs/firewood/commit/ab31227c1e22931b76a24e2eeb98b4c961eef2fa